src/app/components/data-field-edit-dialog/data-field-edit-dialog.component.html

```diff
@@ -47,7 +47,7 @@
     <div class="single-detail">
       <mat-form-field class="full-width">
         <mat-select placeholder="Data Type" [value]="data.dataField.dataType" (selectionChange)="changeDataFieldDataType($event.value)" [disabled]="!data.isContentDeveloper && !data.isOwner">
-          <mat-option *ngFor="let dataFieldType of data.dataFieldTypes" value="{{ dataFieldType }}">{{ dataFieldType }}</mat-option>
+          <mat-option *ngFor="let dataFieldType of (data.dataFieldTypes | keyvalue)" value="{{ dataFieldType.key }}">{{ dataFieldType.key }}</mat-option>
```

Comment:
I am mistified about the type 'keyvalue'. Where is that type defined?

Reply:
It's one of Angular's built-in [pipe](https://angular.dev/guide/pipes) operators. 

https://netbasal.com/introducing-the-new-keyvaluepipe-in-angular-435ac2205c72

Reply:
Thanks, Andrew! The magic is now explained. :)